src/code/z_message_PAL.c

```diff
@@ -1873,14 +1873,14 @@ void Message_StartOcarina(PlayState* play, u16 ocarinaActionId) {
     }
 }
 
-void func_8010BD58(PlayState* play, u16 ocarinaActionId) {
-    play->msgCtx.unk_E40E = 0;
-    Message_StartOcarina(play, ocarinaActionId);
+void Message_StartOcarina(PlayState* play, u16 ocarinaActionId) {
```

Comment:
I might be overthinking this, but from my testing, this is moreso starting the ocarina staff that's covered by the message system instead of starting the ocarina itself. So I was brainstorming `Message_StartOcarinaStaff` or `Message_DisplayOcarinaStaff`. Although this does break down for `OCARINA_ACTION_FREE_PLAY` where you play the ocarina without a staff, so I'm not totally sure.

For context, playing the ocarina needs 3 difference systems to be in sync: player (for stopping motion and pulling out the ocarina), message (for displaying the staff and processing song effects) and audioOcarina (for processing notes/buttons and identifying when a song is played or a mistake is made).

But the actual start of the ocarina, `AudioOcarina_Start`, isn't done through `Message_StartOcarinaImpl`, it's done through `Message_DrawMain` when processing the ocarina states, but is indeed handled by the message system as a whole.

There's also this block of code below in `Message_StartOcarinaImpl` that indicates that this can be called at the end of the ocarina when a song is played by calling `Message_StartOcarina(play, 0x893);` although this function is never called this way, and the text "You played the [song name]" is displayed in a different way.
```c
if (ocarinaActionId >= 0x893) {
    Message_OpenText(play, ocarinaActionId); // You played the [song name]
    textId = 0x86E + ocarinaActionId;
```

Reply:
this is all a pretty fair assessment. And even in my description of this PR I referred to it as more the staff than anything else.
I tbh did not read a single line of the impl function, and was more interested in naming the wrappers and the suns song related part

we could go with this name now (after some more discussion, cause idk if it fits yet), or we could make the change in a PR that is dedicated to this change and makes sure that it is made consistently throughout the system (if more changes are needed)

Reply:
like for example thing that if this changes, the "ocarina action" as a whole probably needs to be reconsidered. but its tough cause `OCARINA_ACTION_CHECK_SARIA` and all the "check" actions sound like a staff handled-thing while `OCARINA_ACTION_FREE_PLAY` doesnt involve any staff (or maybe it does under the hood? lol)

eitherway it seems like more of an involved change than this was meant to be

Reply:
Yeah, the whole `ocarina message` and `ocarina audio` needs another pass because the naming between them is out of sync (I didn't understand the message system at the time of documenting `AudioOcarina` admittedly). 

Also, the check action does go down into the `AudioOcarina` through `ocarinaFlags` (which don't have defines yet), and and differs from message "playback" in whether the ocarina will error after making 1 wrong note (playback) or will only error after filling the whole staff with 8 notes (check).

As you mentioned, this quickly grows out of scope, and these being wrappers to `Message_StartOcarina` and the actual function being `Impl` is a change I do agree with. So on that premise, `Message_StartOcarina` and `Message_StartOcarinaImpl` are fine for this PR